platform/pc/boot/stage1.s

```diff
@@ -75,11 +75,40 @@ dap:
         .lba          dq 1
 
 
+%ifdef DEBUG
+bits 16
+%include "debug.inc"
+bits 32
+%endif
+
 readsectors:
+        mov eax, [dap.sector_count]
+        cmp eax, 0x80
+        jle loop
+        mov ecx, 0x80
+        mov [dap.sector_count], ecx
+loop:
         mov si, dap
         mov ah, 0x42
         mov dl, 0x80
         int 0x13
+%ifdef DEBUG
+        jc sector_read_error
+%endif
+        sub eax, [dap.sector_count]
+        cmp eax, 0
+        jle done 
+        mov ecx, [dap.sector_count]
+        imul ecx, sectorsize
+        add [dap.offset], ecx
+        mov ecx, 0x80
```

Comment:
this should be the minimum of 0x80 and [dap.sector_count]

Reply:
Sorry, I meant to say [dap.sector_count] should be updated with the minimum of 0x80 and eax.